src/client/dbus/thread_api_dbus.cpp

```diff
@@ -0,0 +1,510 @@
+/*
+ *    Copyright (c) 2019, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <assert.h>
+#include <map>
+#include <string.h>
+
+#include "client/dbus/thread_api_dbus.hpp"
+#include "common/code_utils.hpp"
+#include "dbus/constants.hpp"
+#include "dbus/dbus_message_helper.hpp"
+
+using otbr::dbus::CheckErrorMessage;
+using otbr::dbus::DBusMessageEncode;
+using otbr::dbus::DBusMessageExtract;
+using otbr::dbus::DBusMessageToTuple;
+
+static otDeviceRole NameToDeviceRole(const std::string &aRoleName)
+{
+    static std::pair<std::string, otDeviceRole> sRoleMap[] = {
+        {DISABLED_ROLE_NAME, OT_DEVICE_ROLE_DISABLED}, {DETACHED_ROLE_NAME, OT_DEVICE_ROLE_DETACHED},
+        {CHILD_ROLE_NAME, OT_DEVICE_ROLE_CHILD},       {ROUTER_ROLE_NAME, OT_DEVICE_ROLE_ROUTER},
+        {LEADER_ROLE_NAME, OT_DEVICE_ROLE_LEADER},
+    };
+
+    for (auto &&p : sRoleMap)
+    {
+        if (p.first == aRoleName)
+        {
+            return p.second;
+        }
+    }
+
+    assert(false);
```

Comment:
Should not `assert`. Because `aRoleName` seems to be a string decoded from a dbus message. An incoming message should be crash our program.

Suggest returning an error for this case.